src/ripple/consensus/Consensus.h

```diff
@@ -700,6 +718,15 @@ Consensus<Adaptor>::peerProposalInternal(
 
     if (newPeerProp.prevLedger() != prevLedgerID_)
     {
+        uint32_t ourSeq = previousLedger_.seq();
+        if (newPeerProp.proposeSeq() < ++ourSeq)
```

Comment:
This is mixing the sequence number of a `Ledger` and a `Proposal`, which are two different things and are not comparable. 

Proposals are part of the deliberation phase of consensus and indicate which transactions a node believes belong in the next ledger. A sequence of such proposals (each identified by the increasing `proposeSeq`) eventually converge on a final set of transactions, which are used to generate the next ledger.

The ledger sequence number identifies the ledger index, which is one more than the parent ledger number.

Reply:
Aside: Note that the compiler errors are picking up on this difference, because the simulation framework uses tagged integer types to distinguish different kinds of sequence numbers.

Reply:
so why do we not always use tag integer for such integers?

Reply:
The types used in the production code were in place before the consensus code refactor and are used in lots of other parts of the code. I think using tagged integers for them would be great, but would be a decent sized task by itself.